src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -522,14 +528,59 @@ reducer = reduceReducers(
         flowsByName: doDeleteFlow({ name, flowsByName: state.flowsByName })
       }),
 
+      [requestRefreshCallerFlows]: (state: FlowReducer, { payload }) => {
+        const activeFlow = payload || state.currentFlow
+        const currentFlow = state.flowsByName[activeFlow]
+
+        const outcomeNodes = currentFlow.nodes.filter(x => ['success', 'failure'].includes(x.type))
+        const outcomes = _.orderBy(outcomeNodes, 'type', 'desc').map(x => ({
+          condition: `lastNode=${x.name}`,
+          caption: x.friendlyName || x.name,
+          node: ''
+        }))
+
+        const updatedFlows = {}
+        for (const [name, flow] of Object.entries(state.flowsByName)) {
+          if (!flow.nodes.find(n => n.flow === activeFlow)) {
+            continue
+          }
+
+          const updatedFlow = {
+            ...flow,
+            nodes: flow.nodes.map(node => {
+              if (node.flow !== activeFlow) {
+                return node
+              }
+
+              return {
+                ...node,
+                next: outcomes.map(o => ({ ...o, node: node.next.find(n => n.condition === o.condition)?.node || '' }))
+              }
+            })
+          }
+
+          updatedFlows[name] = updatedFlow
```

Comment:
I got confused between updatedFlows and updatedFlow, you could make the operation clearer this way: 

```js
let updatedFlows = {}
// ...
updatedFlows = {
  ...updatedFlows,
  [name]: {
    ...flow, nodes: []
  }
}

```